docs/book/products/catalog_promotions.rst

```diff
@@ -313,6 +313,27 @@ After changes in CatalogPromotion, we dispatch proper message with delay calcula
     To enable asynchronous Catalog Promotion, remember about running messenger consumer in a separate process, use the command: ``php bin/console messenger:consume main``
     For more information check official `Symfony docs <https://symfony.com/doc/current/messenger.html#consuming-messages-running-the-worker>`_
 
+Catalog promotion synchronicity
+-------------------------------
+
+Tl;dr - there is no reason to use synchronous processing unless you have only a few products and manually activate and deactivate catalog promotions.
+
+Since asynchronous transport is the default setting, to use synchronous processing, you need to update your `.env` configuration file:
+
+.. code-block:: bash
+
+    MESSENGER_TRANSPORT_DSN=sync://
+
+Details:
+''''''''
+
+- synchronous processing in opposite to asynchronous means that after confirmation the administrator will wait for a server response until processing is complete
+- this may cause a bad user experience if there are other promotions as it causes them to be recalculated as well
+- it is not possible to create a scheduled catalog promotion that starts automatically when the start date is reached, it has to be manually started
+- if there is a catalog promotion that is within the timeframe but has not been applied for the reason described above, it will be triggered when any action related to creating or editing any promotion is performed
+- if the validity period of the promotion has expired, the promotion will not be automatically inactive
```

Comment:
```suggestion
- The same happens with the end date, so if the catalog promotion should have expired, it will not become inactive automatically, inactivation has to be triggered manually.
```